src/controller/domain_handle.rs

```diff
@@ -186,6 +189,19 @@ impl DomainHandle {
         self.shards[shard].worker.clone()
     }
 
+    pub fn assigned_to_worker(&self, worker: &WorkerIdentifier) -> bool {
```

Comment:
This isn't the cleanest API, though I'm not sure of what might be better

Reply:
I think it's fine, at least for now. In the future, we might want to obtain the shard ID as a return value to selectively recovery that shard only.